app/models/user.rb

```diff
@@ -3,9 +3,10 @@ class User < ApplicationRecord
   has_many :parties, through: :user_parties
   has_many :friends, class_name: "UserFriend", foreign_key: "user_id"
 
-  validates :email, uniqueness: true, presence: true
-  validates :password, presence: true
+  validates :email, uniqueness: true, presence: true, format: /\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/
+  validates :password, length: { minimum: 8 }, presence: true
+  # validates :password_confirmation, presence: { message: "failed, passwords must match. Please try again." }
+  validates_confirmation_of :password
```

Comment:
I have a potential fix for this in the future, we can circle back to this with fresh eyes